pkg/actuators/machine/vm.go

```diff
@@ -62,38 +74,60 @@ func createVM(mscp *machineScope, userData []byte) (*nutanixClientV3.VMIntentRes
 				Kind: utils.StringPtr("image"),
 				UUID: imageUuidPtr,
 			},
-			DiskSizeMib: utils.Int64Ptr(mscp.providerSpec.DiskSizeMib),
+			DiskSizeMib: utils.Int64Ptr(GetMibValueOfQuantity(mscp.providerSpec.SystemDiskSize)),
 		})
+
 		vmMetadata := nutanixClientV3.Metadata{
 			Kind:        utils.StringPtr("vm"),
 			SpecVersion: utils.Int64Ptr(1),
 		}
+
+		// Add the category for installer clueanup the VM at cluster torn-down time
+		// if the category exists in PC
+		err = addCategory(mscp, &vmMetadata)
+		if err != nil {
```

Comment:
Nit, i would inline this so we don't have to reuse the previous variable
```suggestion
		if err := addCategory(mscp, &vmMetadata); err != nil {
```